apps/store/themes/store/js/categorization.js

```diff
@@ -204,10 +209,37 @@ $(function() {
         resetPageAttributes();
         store.infiniteScroll.addItemsToPage();
 
-        var searchQuery =  url.split("q=")[1];
-        $('#search').val(formatSearchQuery(decodeURIComponent(searchQuery)));
+        setCategorizationQuery(url);
+    };
+
+    var setCategorizationQuery = function(url){
+        var searchQuery =  removeUnrelatedKeys(decodeURIComponent(url));
+        $('#categorization-query').val(formatSearchQuery(searchQuery));
     };
 
+    var removeUnrelatedKeys = function(url){
```

Comment:
This method removes the keys from the search query which are not related to taxonomy for categorization in order to make generic search independent from taxonomy